src/crimson/osd/osd_operations/client_request.cc

```diff
@@ -221,55 +217,44 @@ ClientRequest::process_op(instance_handle_t &ihref, Ref<PG> &pg)
   }).then_interruptible([this, pg, &ihref]() mutable {
     return pg->already_complete(m->get_reqid()).then_interruptible(
       [this, pg, &ihref](auto completed) mutable
-      -> PG::load_obc_iertr::future<seq_mode_t> {
+      -> PG::load_obc_iertr::future<> {
       if (completed) {
         auto reply = crimson::make_message<MOSDOpReply>(
           m.get(), completed->err, pg->get_osdmap_epoch(),
           CEPH_OSD_FLAG_ACK | CEPH_OSD_FLAG_ONDISK, false);
 	reply->set_reply_versions(completed->version, completed->user_version);
         return conn->send(std::move(reply)).then([] {
-          return seastar::make_ready_future<seq_mode_t>(seq_mode_t::OUT_OF_ORDER);
+          return seastar::make_ready_future<>();
         });
       } else {
         return ihref.enter_stage<interruptor>(pp(*pg).get_obc, *this
 	).then_interruptible(
-          [this, pg, &ihref]() mutable -> PG::load_obc_iertr::future<seq_mode_t> {
+          [this, pg, &ihref]() mutable -> PG::load_obc_iertr::future<> {
           logger().debug("{}: got obc lock", *this);
           op_info.set_from_op(&*m, *pg->get_osdmap());
-          // XXX: `do_with()` is just a workaround for `with_obc_func_t` imposing
-          // `future<void>`.
-          return seastar::do_with(
-	    seq_mode_t{},
-	    [this, &pg, &ihref](seq_mode_t& mode) {
-	      return pg->with_locked_obc(
-		m->get_hobj(), op_info,
-		[this, pg, &mode, &ihref](auto obc) mutable {
-		  return ihref.enter_stage<interruptor>(pp(*pg).process, *this
-		  ).then_interruptible(
-		    [this, pg, obc, &mode, &ihref]() mutable {
-		      return do_process(ihref, pg, obc
-		      ).then_interruptible([&mode] (seq_mode_t _mode) {
-			mode = _mode;
-			return seastar::now();
-		      });
-		    });
-		}).safe_then_interruptible([&mode] {
-		  return PG::load_obc_iertr::make_ready_future<seq_mode_t>(mode);
-		});
-	    });
-	  });
+          return pg->with_locked_obc(
+            m->get_hobj(), op_info,
+            [this, pg, &ihref](auto obc) mutable {
+              return ihref.enter_stage<interruptor>(pp(*pg).process, *this
+            ).then_interruptible([this, pg, obc, &ihref]() mutable {
+              return do_process(ihref, pg, obc);
+            });
+          });
+        }).safe_then_interruptible([] {
+          return PG::load_obc_iertr::make_ready_future<>();
+        });
       }
-      });
-  }).safe_then_interruptible([pg] (const seq_mode_t mode) {
-    return seastar::make_ready_future<seq_mode_t>(mode);
+    });
+  }).safe_then_interruptible([pg] {
+    return seastar::make_ready_future<>();
```

Comment:
how about just replacing `safe_then_interruptible()` with `handle_error_interruptible()`, like

```suggestion
  }).handle_error_interruptible([pg, this] {
    PG::load_obc_ertr::all_same_way([this, pg=std::move(pg)](const auto &code) {
      // ...
  }));  
```

Reply:
https://github.com/ceph/ceph/pull/49127/files#diff-3465c956b7a8e8fab614e23efef1c6093d8277c66ac36a78a138040010a92eceR246-R250